### PR TITLE
Add more exceptions

### PR DESCRIPTION
**Pull request by Keith-CY**
Add custom exceptions to make code neat.

### CODE REVIEW

File: packages/neuron-wallet/src/controllers/transactions.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import Key from '../keys/key'
 
 import { Controller as ControllerDecorator, CatchControllerError } from '../decorators'
 import { Channel, ResponseCode } from '../utils/const'
-import i18n from '../utils/i18n'
+import { KeyHasNotData, TransactionIsNotFound, CurrentWalletIsNotSet, ServiceHasNotResponse } from '../exceptions'
```
Comment (classicalliu) [inline packages/neuron-wallet/src/controllers/transactions.ts]: `ServiceHasNoResponse` or `ServiceNotRespond`

File: packages/neuron-wallet/src/exceptions/key.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import i18n from '../utils/i18n'
+
+export class KeyHasNotData extends Error {
```
Comment (classicalliu) [inline packages/neuron-wallet/src/exceptions/key.ts]: `KeyHasNoData`

File: packages/neuron-wallet/src/exceptions/key.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  }
+}
+
+export class UnsupportedCipher extends Error {
+  constructor() {
+    super('messages.unsupported-cipher')
+  }
+}
+
+export class MnemonicIsInvalid extends Error {
```
Comment (classicalliu) [inline packages/neuron-wallet/src/exceptions/key.ts]: `MnemonicInvalid`

File: packages/neuron-wallet/src/exceptions/network.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+export class DefaultNetworkIsUnremovable extends Error {
+  constructor() {
+    super(i18n.t('messages.default-network-is-unremovable'))
+  }
+}
+
+export class LackOfDefaultNetork extends Error {
```
Comment (classicalliu) [inline packages/neuron-wallet/src/exceptions/network.ts]: `LackOfDefaultNetwork`

File: packages/neuron-wallet/src/exceptions/transaction.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import i18n from '../utils/i18n'
+
+export class TransactionIsNotFound extends Error {
```
Comment (classicalliu) [inline packages/neuron-wallet/src/exceptions/transaction.ts]: `TransactionNotFound`

File: packages/neuron-wallet/src/exceptions/wallet.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import i18n from '../utils/i18n'
+
+export class CurrentWalletIsNotSet extends Error {
```
Comment (classicalliu) [inline packages/neuron-wallet/src/exceptions/wallet.ts]: I this these `Is` can be removed

Comment (ashchan) [reply packages/neuron-wallet/src/exceptions/wallet.ts]: +1

File: packages/neuron-wallet/src/exceptions/network.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import i18n from '../utils/i18n'
+
+export class NetworkNotFound extends Error {
+  constructor(id: string) {
+    super(i18n.t('messages.network-is-not-found', { id }))
```
Comment (classicalliu) [inline packages/neuron-wallet/src/exceptions/network.ts]: Should we also remove `is` here ?

**Pull request merged**